ui/src/app/shared/services/extensions-service.ts

```diff
@@ -11,6 +11,7 @@ export interface Extension {
 export interface ExtensionComponentProps {
     node: ResourceNode;
     tree: ApplicationTree;
+    state: {spec: State};
```

Comment:
The `State` interface represents K8S resource ( the interface name is pretty confusing ). So `state` type should be just `State` (not `{spec: State}` ).


I've realized it was a mistake to add `node` field. The extension only need live resource. Let's change `ExtensionComponentProps` as following:

```
export interface ExtensionComponentProps {
    tree: ApplicationTree;
    resource: State;
}
```